### PR TITLE
[#155911635] Add sonar ts

### PR DESCRIPTION
**Pull request by lussoluca**
Add sonar ts and fix lint issues.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#155911635](https://www.pivotaltracker.com/story/show/155911635): Integrare sonar-ts nel processo di linting

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/controllers/authenticationController.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export default class AuthenticationController {
     const errorOrUser = validateSpidUser(userPayload);
 
     errorOrUser.fold(
+      // tslint:disable-next-line:no-identical-functions
       (error: Error) => {
```
Comment (cloudify) [inline src/controllers/authenticationController.ts]: this one is really identical, you could refactor it with:

```ts
const return500Error = (error: Error) => {
         res.status(500).json({
           message: error.message
         });
       }
```

File: src/services/profileService.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export default class ProfileService {
     upsertProfile: ExtendedProfile
   ): Promise<ProfileWithEmail> {
     return new Promise(async (resolve, reject) => {
+      // tslint:disable-next-line:no-use-of-empty-return-value await-promise
```
Comment (cloudify) [inline src/services/profileService.ts]: I'm not sure why this rule is triggered, does `upsertProfile` return a value?

Comment (lussoluca) [reply src/services/profileService.ts]: yes upsertProfile returns a Promise in the same way as `getProfile` at line 32 in the same file, but sonar mark only this as error...

Comment (cloudify) [reply src/services/profileService.ts]: I think there is a problem here with the fact that you're passing an async function ti the `Promise` constructor (AFAIK you can't)

**Pull request merged**